automation/label-approved-veps.py

```diff
@@ -0,0 +1,118 @@
+import os
+import re
+import requests
+
+GITHUB_TOKEN = os.environ["GITHUB_TOKEN"]
+KUBEVIRT_REPO = os.environ.get("GITHUB_REPOSITORY", "kubevirt/kubevirt")
+PR_NUMBER = os.environ["PR_NUMBER"]
+HEADERS = {"Authorization": f"token {GITHUB_TOKEN}", "Accept": "application/vnd.github.v3+json"}
+
+def get_pr_details():
+    #Fetch the kubevirt/kuebvirt PR body
+    url = f"https://api.github.com/repos/{KUBEVIRT_REPO}/pulls/{PR_NUMBER}"
+    response = requests.get(url, headers=HEADERS)
+    response.raise_for_status()
+    return response.json()["body"]
+
+def extract_enhancements_references(pr_body):
+    #Get the enhancements reference numbers from the PR.
+    pattern = r"(?:https://github.com/)?kubevirt/enhancements/(?:issues|pull)/(\d+)|(?:kubevirt/)?enhancements#(\d+)"
+    matches = re.findall(pattern, pr_body)
+
+    # Extract the first non-empty group from each match
+    ref_numbers = [group[0] or group[1] for group in matches]
+    # there can be duplicates
+    return list(set(ref_numbers))
+
+def is_merged_pr(ref_number):
+    url = f"https://api.github.com/repos/kubevirt/enhancements/pulls/{ref_number}"
+    response = requests.get(url, headers=HEADERS)
+    if response.status_code == 200:
+        pr_data = response.json()
+        return pr_data.get("merged_at") is not None
+    return False
+
+def find_related_merged_prs(issue_number):
+    #Find merged PRs in kubevirt/enhancements that reference a given issue.
+    
+    # Construct the search query for merged PRs referencing the issue
+    query = (
+        f'repo:kubevirt/enhancements is:pr is:merged '
```

Comment:
Shouldn't you be looking for issues instead of PRs? Or did I misunderstand your comment about `force users to reference only the issue`?

Reply:
The users reference a vep issue. We need to look for merged VEP PR's - that will tell that the issue is approved therefore this pr is related to an approved VEP. 

This will need a follow up so we will distinguish between releases.

Reply:
> This will need a follow up so we will distinguish between releases.

Kubernetes use the `/milestone` label. Perhaps we can use it, then apply the label only to the latest milestone (being v1.6 currently)